src/providers/tfc/planProvider.ts

```diff
@@ -0,0 +1,424 @@
+              planLog.diagnosticSummary.warningCount += 1;
+            }
+            if (logLine.diagnostic.severity === 'error') {
+              planLog.diagnosticSummary.errorCount += 1;
+            }
+            continue;
+          }
+
+          // TODO: logLine.type=test_*
+        } catch (e) {
+          // skip any non-JSON lines, like Terraform version output
+          continue;
+        }
+      }
+
+      return planLog;
+    } catch (error) {
+      let message = `Failed to obtain plan from ${plan.logReadUrl}: `;
+
+      if (error instanceof ZodiosError) {
+        handleZodiosError(error, message, this.outputChannel, this.reporter);
+        return;
+      }
+
+      if (axios.isAxiosError(error)) {
+        if (error.response?.status === 401) {
+          handleAuthError();
+          return;
+        }
+
+        if (isErrorFromAlias(apiClient.api, 'listRuns', error)) {
+          message += apiErrorsToString(error.response.data.errors);
+          vscode.window.showErrorMessage(message);
+          this.reporter.sendTelemetryException(error);
+          return;
+        }
+      }
+
+      if (error instanceof Error) {
+        message += error.message;
+        vscode.window.showErrorMessage(message);
+        this.reporter.sendTelemetryException(error);
+        return;
+      }
+
+      if (typeof error === 'string') {
+        message += error;
+      }
+      vscode.window.showErrorMessage(message);
+      return;
+    }
+  }
+
+  dispose() {
+    //
+  }
+}
+
+interface PlanLog {
+  plannedChanges?: Change[];
+  changeSummary?: ChangeSummary;
+  driftChanges?: Change[];
+  driftSummary?: DriftSummary;
+  outputs?: Outputs;
+  diagnostics?: Diagnostic[];
+  diagnosticSummary?: DiagnosticSummary;
+}
+
+interface ItemWithChildren {
+  getChildren(): vscode.TreeItem[];
+}
+
+function isItemWithChildren(element: object): element is ItemWithChildren {
+  return 'getChildren' in element;
+}
+
+class PlannedChangesItem extends vscode.TreeItem implements ItemWithChildren {
+  constructor(private plannedChanges: Change[], summary?: ChangeSummary) {
+    let label = 'Planned changes';
+    if (summary) {
+      const labels: string[] = [];
+      if (summary.import > 0) {
+        labels.push(`${summary.import} to import`);
+      }
+      if (summary.add > 0) {
+        labels.push(`${summary.add} to add`);
+      }
+      if (summary.change > 0) {
+        labels.push(`${summary.change} to change`);
+      }
+      if (summary.remove > 0) {
+        labels.push(`${summary.remove} to destroy`);
+      }
+      if (labels.length > 0) {
+        label = `Planned changes: ${labels.join(', ')}`;
+      }
+    }
+    super(label, vscode.TreeItemCollapsibleState.Expanded);
+  }
+
+  getChildren(): vscode.TreeItem[] {
+    const items: vscode.TreeItem[] = [];
+    this.plannedChanges.forEach((change) => {
+      items.push(new PlannedChangeItem(change));
+    });
+    return items;
```

Comment:
```suggestion
    return this.plannedChanges.map((change) => new PlannedChangeItem(change));
```
`map()` is a good way to transform list items.

You could apply this to the other `getChildren` functions, too.